File: LeetCode/Easy/0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {

        if(s.length() == 0) return true;
        if(t.length() == 0) return false;

        int idx = 0;
        char[] sc = s.toCharArray();

        for (char tc : t.toCharArray()) {
            if (sc[idx] == tc) {
                idx++;
                if (idx >= sc.length) return true;
            }
        }
        return false;
    }
}

File: LeetCode/Easy/0392-is-subsequence/0392-is-subsequence.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        if(s.length == 0) return true;
        if(t.length == 0) return false;

        var idx = 0;

        for (tc in t.toCharArray()) {
            if (s[idx] == tc) {
                idx++;
                if(idx >= s.length) return true;
            }
        }
        return false;
    }
}